### ISSUE TITLE
Add new and update form list [v3 - Discussion]

### ISSUE THREAD

**Issue by rowo** [onaio/collect#24]
Notes from meeting:

Main user problem: field enumerators and users aren’t notified if form is changed or if there is a new form, so they are filling out old forms.

Considerations: forms shouldn't be updated automatically bc this can cause problems

Initial proposal: Add a “New and updated forms” button. Problem is this includes all "new to phone" forms, not just the "new to user" forms, meaning forms will show up in the list as long as they haven't been downloaded to the phone.

**Comment by rowo** [comment]
- With the the button, I think we can handle it one of two ways: A. add the new button, or B, add new/updated forms to the Blank form list. A probably makes the most sense for being training-friendly and follows what I would expect. It would be nice to have a number (in red) to show if there's anything inside:

<img src="https://cloud.githubusercontent.com/assets/1584163/11790371/654bbfce-a269-11e5-8563-ebf221800915.png" width="250">
- It would also be nice to have an alert on existing blank forms that have been replaced:

<img src="https://cloud.githubusercontent.com/assets/1584163/11790399/8a8b6c1c-a269-11e5-8074-81ec1fe8219c.png" width="250">
- The new and updated form list could be nice with "added" or "updated" information in each row. It would be great to hide forms and not show everything New that the phone hasn't downloaded. For "new" forms, do we do hide with a button to dismiss, expiration time (after seeing), or after the button and page is seen once? Dismiss is safest, though the problem a list could need a lot of a lot of initial editing. One view is dangerous if a user misses it. iPhone notifications is like expiration time (with dismiss).

<img src="https://cloud.githubusercontent.com/assets/1584163/11790667/fc4e5cfa-a26a-11e5-9317-97fe74f827e0.png" width="250">

**Comment by imwongela** [comment]
To hide forms on the new list we can combine the two methods pointed (expiration time + dismiss button). The user will have the option to hide a form. The expiration time can be number of views(maybe ~5 views) and if the user hasn't acted it can be hidden. It also solves the problem of displaying as new any form the user hasn't downloaded.

**Comment by rowo** [comment]
Yeah, I was thinking more of a real expiration time like two weeks from being shown, but number of times could work too.

@mberg @smusula @jmunene would be good to hear your voices in this discussion too.

**Comment by imwongela** [comment]
Ok, real time will be better especially when a user opens the app often

**Comment by mberg** [comment]
Just went through this with Isaac and Koros.

Think we decided to:
- only show forms that are 30 days old or younger (we can do 2 weeks instead but I think 30 might make sense)
- Allow the user to dismiss the form.   They can always go to get new forms if they want to download it.

**Comment by mberg** [comment]
@rowo it would be nice if we could have a small updated or new tag/badge next to the name of the form instead of the text (updated)

**Comment by rowo** [comment]
@mberg Do tags or badges exist in ODK Collect currently? It could be a background color with reversed out type on top. I am hesitating on adding too much new visual design or ux patterns just yet.

**Comment by rowo** [comment]
Would dismiss also dismiss the tag near forms that have been updated? Probably not I guess.

**Comment by rowo** [comment]
Made updates the screen based on conversation above:

Home screen notification in a box and uses Ona color.
<img src="https://cloud.githubusercontent.com/assets/1584163/11907187/f475a0d2-a59f-11e5-87c1-c9fb9b7a084e.png" width="300">

Changed "Shared Forms" to "Form Management" in General Settings and added setting for expiration time for new/updated forms.
<img src="https://cloud.githubusercontent.com/assets/1584163/11907188/f68fe58a-a59f-11e5-9503-ad8b24fab118.png" width="300">

Added a "Dismiss Selected" button. For small screens, the buttons can stack. Or we can change the UI much more by hiding the Get and Dismiss buttons until a box is actually selected and hide Toggle All after a box is selected. I don't think we should change the UI so much yet. Any other ideas?
<img src="https://cloud.githubusercontent.com/assets/1584163/11907191/fa6da340-a59f-11e5-986f-86cc6fb4acd4.png" width="300">

Made the "Updated form" into a text tag. I think this is persistent even if the updated form notification is dismissed in the New/Updated Form list.
<img src="https://cloud.githubusercontent.com/assets/1584163/11907197/fe4e71ce-a59f-11e5-8e06-b2d78eccee3d.png" width="300">

**Comment by mberg** [comment]
These look good. 

I agree we don't want to change too much what ODK provides for the moment.  No badges don't exist. If it is too difficult we can just add data in a parenthesis.

 @imwongela do you have what you need to do this?

I've add a v3 label (since the current milestones are confusing) to identify which issues need to be worked on.

**Comment by imwongela** [comment]
The label is ok @mberg. Also I have noted there are forms with very long names and also after addition of a dismiss button it looks a bit compressed. @rowo how should this look considering the above? Is the label for updated forms only or will the new also have a "New" label?

**Comment by rowo** [comment]
@mberg it's not about technical difficulty. It's about not wanting to change to much from a user experience standpoint.

@imwongela I don't think there is anything we can do with very long names. The label just goes after the name no matter the length. We can add a new label, though I don't think it's as important as the "Update available" one right now and it would probably disappear after viewed once. 

I don't know what to do about the dismiss button making it cramped. I think we want to keep it consistent so people trained in using ODK Collect can use it right now. I explained my thinking above, so if there are no other ideas...

**Comment by imwongela** [comment]
@rowo  the label is breaking the layout when the names are long. If the name extends the full length of the screen, label wont appear and this causes additional space after the formId.

**Comment by rowo** [comment]
@imwongela are you saying there's no way around this? What are the alternatives? If we use a corner flag or icon, would that help?
